docs/kilnjs.md

```diff
@@ -66,9 +66,81 @@ You can attach events to the schema inputs. The events you can attach vary depen
   });
 ```
 
-### Validation
+## Validation
 
-By attaching events, one of the things you can use kiln.js for is to do field validation.
+There are several strategies that can be used to validate fields and forms.
+
+#### Custom validators in the Schema file
+You can simply rely on the current custom validators and use them normally in your schema.yml file.  kiln.js will not affect their behavior and they will continue to function as normal.
+
+#### kilnjs schema.validation
+You can combine the power of custom validators with kiln.js by adding a validation property to the schema that specifies which field should be validated using which custom validator.
+```js
+  schema.validation = [
+    {
+      input: 'title', // the name of the field to be validated
+      validator: 'penelopeCruz' // the name of the custom validator to validate the field with
+    }
+  ];
+```
+In the custom validator file that lives in the services/kiln/validators folder, you construct a validator file in much the same way you would create one to be used by the schema file, only instead of exporting a function named validate, you export one called kilnjsValidate that validates on the value you pass it.
+```js
+import _ from 'lodash';
+const { forEachComponent, forEachField, getPlaintextValue, getPreviewText, labelComponent }= window.kiln.utils.validationHelpers;
+
+export const label = 'Penélope Cruz',
+  description = 'Penélope Cruz should always be spelled with an accent mark',
+  type = 'warning';
+
+
+function containsPenelopeCruz(value) {
+  return _.includes(getPlaintextValue(value).toLowerCase(), 'penelope cruz');
+}
+
+export function kilnjsValidate(value) {
+  const text = getPlaintextValue(value),
+    index = text.toLowerCase().indexOf('penelope cruz');
+
+  return containsPenelopeCruz(value)
+    ? { label, description, type, preview: getPreviewText(text, index, 13) }
+    : false;
+}
+```
+Errors or warnings generated by kilnjsValidate appear on the health tab rather than giving immediate feedback, in the same way that ordinary custom validators work.
+
+You can create custom validators that validate based on the value of multiple fields as well as single fields.  If validation rules depend upon multiple fields, you pass an array of fields in the input property.  Just remember that the order of field names in the array is important because the custom validator function will recieve an array with the value of those fields as arguments in the same order as they appear in the original array.
```

Comment:
Maybe adding Warning signs to the remember phrase or showing it as an alert could add some value

Reply:
👍